syntax = "proto3";

package dotaservice;

import "protobuf/CMsgBotWorldState.proto";
import "protobuf/DotaAction.proto";

message Action {
  DotaAction action = 1;
  string dota_time = 3; // should be float
}

message Observation {
  CMsgBotWorldState world_state = 1;
}

message Config {
  // Intentionally empty.
}

service DotaService {
  rpc reset (Config) returns (Observation) {}
  rpc step (Action) returns (Observation) {}
}
